cmd/admission-controller/mutatingwebhook.go

```diff
@@ -0,0 +1,150 @@
+type recommendationApplier struct {
+	Client  client.Client
+	decoder *admission.Decoder
+
+	c      client.Client
+	logger log.Logger
+}
+
+func getDataCenterRecommendations(sca *v1alpha1.ScyllaClusterAutoscaler) []v1alpha1.DataCenterRecommendations {
+	if sca.Status.Recommendations == nil {
+		return nil
+	}
+	dcRecs := sca.Status.Recommendations.DataCenterRecommendations
+	if len(dcRecs) == 0 {
+		return nil
+	}
+	return dcRecs
+}
+
+func getRackRecommendations(dataCenterName string,
+	dcRecs []v1alpha1.DataCenterRecommendations) []v1alpha1.RackRecommendations {
+	for idx := range dcRecs {
+		if dcRecs[idx].Name == dataCenterName {
+			return dcRecs[idx].RackRecommendations
+		}
+	}
+	return nil
+}
+
+func findRack(rackName string, racks []scyllav1.RackSpec) *scyllav1.RackSpec {
+	for idx := range racks {
+		if rackName == racks[idx].Name {
+			return &racks[idx]
+		}
+	}
+	return nil
+}
+
+func mutateCluster(ctx context.Context, logger log.Logger, cluster *scyllav1.ScyllaCluster, c client.Client) error {
+	logger.Info(ctx, "Starting mutation of ScyllaCluster")
+	scas := &v1alpha1.ScyllaClusterAutoscalerList{}
+	if err := c.List(ctx, scas); err != nil {
+		return fmt.Errorf("Failed to get SCAs: %s", err)
+	}
+
+	logger.Debug(ctx, "SCAs fetched", "num", len(scas.Items))
+
+	for idx := range scas.Items {
+		sca := &scas.Items[idx]
+
+		if *sca.Spec.UpdatePolicy.UpdateMode == v1alpha1.UpdateModeOff {
+			logger.Debug(ctx, "Autoscaler has 'off' scaling policy, skipping", "autoscaler", sca.ObjectMeta.Name)
+			continue
+		}
+
+		targetRef := sca.Spec.TargetRef
+		referencedCluster := &scyllav1.ScyllaCluster{}
+		var err = c.Get(ctx, client.ObjectKey{
+			Namespace: targetRef.Namespace,
+			Name:      targetRef.Name,
+		}, cluster)
+		if err != nil {
+			return fmt.Errorf("Fetch referenced ScyllaCluster: %s", err)
+		}
+
+		if referencedCluster != cluster {
+			logger.Debug(ctx, "SCA not pointing to cluster under mutation")
+		}
+
+		logger.Debug(ctx, "Cluster has 'Initial' or 'Auto' scaling policy")
+
+		dcRecs := getDataCenterRecommendations(sca)
+		if dcRecs == nil {
+			logger.Debug(ctx, "No recommendations for cluster", "cluster", cluster.Name)
+			continue
+		}
+
+		dataCenterName := cluster.Spec.Datacenter.Name
+
+		logger.Info(ctx, "Found data center with name", "data center", dataCenterName)
+
+		rackRecs := getRackRecommendations(dataCenterName, dcRecs)
+		if rackRecs == nil {
+			logger.Debug(ctx, "No recommendations for data center", "data center", dataCenterName)
+			continue
+		}
+
+		for j := range rackRecs {
+			rackRec := &rackRecs[j]
+
+			if rackRec.Members == nil {
+				logger.Debug(ctx, "No members recommendation for rack", "rack", rackRec.Name)
+				continue
+			}
+
+			rack := findRack(rackRec.Name, cluster.Spec.Datacenter.Racks)
+			if rack == nil {
+				logger.Debug(ctx, "Could not find rack matching recommendation", "rack", rackRec.Name)
+				continue
+			}
+
+			rack.Members = rackRec.Members.Target
+
```

Comment:
Update is missing, you're modyfing rack in memory of admission controller process.